activation/nipost.go

```diff
@@ -91,19 +105,28 @@ type poetDbAPI interface {
 func NewNIPostBuilder(
 	nodeID types.NodeID,
 	postSetupProvider postSetupProvider,
-	poetProvers []PoetProvingServiceClient,
 	poetDB poetDbAPI,
+	poetServers []string,
```

Comment:
Why changed to `string`? It allowed passing a mocked instance in tests nicely. Now there are 2 ways of passing poet provers and in tests, you had to pass an empty list explicitly. What's the benefit?

Reply:
The service now receives the configuration and decides by itself what to do with it. Imo it is not correct that the node initializes http clients and passes them to the service, when the service should do that itself.

There is still only one way to pass PoET provers: via this parameter. The second one is a functional option that is not exported and thereby only usable in tests (a pattern that we use in other services as well).

Reply:
I'm not sure, it seems to be a step back. The API lost flexibility as now the HTTP interface to Poets is hardcoded in the NipostBuilder (previously it could be any client).